packages/frontend/src/pages/Tile/hooks/useCreateRow.tsx

```diff
@@ -0,0 +1,82 @@
+import { Dispatch, SetStateAction, useCallback, useState } from 'react'
+import { useMutation } from '@apollo/client'
+import { CREATE_ROW } from 'graphql/mutations/create-row'
+
+import { NEW_ROW_ID, TEMP_ROW_ID_PREFIX } from '../constants'
+import { GenericRowData } from '../types'
+
+interface CreateRowInput {
+  input: {
+    tableId: string
+    data: Record<string, string>
+  }
+}
+
+interface CreateRowOutput {
+  createRow: string
+}
+
+export function useCreateRow(
+  tableId: string,
+  setData: Dispatch<SetStateAction<GenericRowData[]>>,
+) {
+  const [rowsCreated, setRowsCreated] = useState<Set<string>>(new Set())
+
+  const [createRowMutation] = useMutation<CreateRowOutput, CreateRowInput>(
+    CREATE_ROW,
+    {
+      onCompleted({ createRow: createdRowId }) {
+        setRowsCreated((prev) => new Set(prev.add(createdRowId)))
+        setTimeout(() => {
+          setRowsCreated((prev) => {
+            prev.delete(createdRowId)
+            return new Set(prev)
+          })
+        }, 1000)
+      },
+    },
```

Comment:
long term, we should probably provide `onError` and handle row creation failures